### PR TITLE
Will/eng 1598/download process data

### PR DESCRIPTION
**Pull request by wbrinkerhoff**

### CODE REVIEW

File: fero/process.py [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+    schema_class = ProcessSchema
+
+    def __repr__(self):
+        return f"<Process name={self.name}>"
+
+    __str__ = __repr__
+
+    @property
+    @memoized(maxsize=1)
```
Comment (lukegrecki) [inline fero/process.py:178]: Why only ` maxsize=1`?

File: fero/process.py [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+    def get_data(
+        self,
+        tags: Sequence[Union[Tag, str]],
+        kpis: Optional[Sequence[Union[Tag, str]]] = None,
+    ) -> pd.DataFrame:
+        """Downloads a pandas data frame with specified process data.
+
+        Returns a pandas data frame consisting of the specified tags and bounded by an optional list of kpis.
+        For all process types specifying a kpi will remove any tags from stages that are after
+        the stage the kpi is assigned too.  For advanced and batch process types not providing a kpi will
```
Comment (lukegrecki) [inline fero/process.py:251]: Typo: `assigned to`.

File: tests/test_client.py [hunk 3]
```diff
@@ -291,3 +292,53 @@ def test_get_datasource_success(patch_fero_get, datasource_data):
...
+    assert len(processes) == 1
+    assert isinstance(processes[0], Process)
+    assert processes[0].name == process_data["name"]
+    patch_fero_get.assert_called_with(
+        "/api/processes/", params={"name": process_data["name"]}
+    )
+
+
+def test_search_processes_paginated(patch_fero_get, process_data):
+    """Test that a list analyses is returned by search_analyses"""
```
Comment (lukegrecki) [inline tests/test_client.py:323]: This should read `processes` not `analyses`.

**Review by lukegrecki — APPROVED** [review]
Some minor typos and one question. Looks good! Nice work!

It is probably time soon to have an integration test with the app in some way, as I can see these schemas starting to drift quite quickly and potentially cause problems for the client.

File: fero/process.py (hunk 1 above)
Comment (wbrinkerhoff) [reply fero/process.py:178]: limits it to only cache 1 value (which should only be the original object).  Probably doesn't matter, but I didn't look to see if the implementation would reserve memory for more or anything....Also, stackoverflow just did it that way.

**Review by lukegrecki — APPROVED** [review]

**Pull request merged**